lib/prerequisite.js

```diff
@@ -1,13 +1,48 @@
 'use strict';
 const execa = require('execa');
 const Listr = require('listr');
+const pTimeout = require('p-timeout');
 const version = require('./version');
 
 module.exports = (input, pkg, opts) => {
 	let newVersion = null;
 	let tagPrefix = 'v';
 
-	const tasks = [
+	const tasks = [{
+			title: 'Ping npm registry',
+			task: () => pTimeout(execa.stdout('npm', ['ping'])
+				.catch(() => {
+					throw new Error('Connection to npm registry failed');
+				}), 6000, 'Connection to npm registry timed out')
+		},
+
+		{
+			title: 'Verify user is authenticated',
+			task: () => execa.stdout('npm', ['whoami'])
+				.catch(err => {
```

Comment:
Having `.catch()` at the end of chain binds listener to `then`'s inner `Promise` as well.

Can't think of alternative.

Reply:
It looks correct now.